pkg/apis/eksctl.io/v1alpha5/types.go

```diff
@@ -484,9 +484,14 @@ type NodeGroup struct {
 
 	// +optional
 	Labels map[string]string `json:"labels,omitempty"`
+	// +optional
+	LabelsOnSpot map[string]string `json:"labelsOnSpot,omitempty"`
```

Comment:
Thanks @jaypipes !. That makes sense, It was also one of the reasons to classify it as a "naive attempt" :) 

I was wondering if the API needs changing at all. For example, I commented on the possibility of opting for sensible defaults; We could implement so that nodes get labelled (label name to be determined) with "Spot" / "On-Demand" by default. That way affinities can be applied (and no API change is required). Taints will require a different consideration however.

As for the API change, I think this other issue #1191 present a stronger case to change the API, for example. I'm not sure how API changes are planned and coordinated on the project.

Will be interesting to get the views on this from @errordeveloper .

Reply:
Ping @errordeveloper :) What are your thoughts about how this PR changes the API objects?